updater/aws.go

```diff
@@ -60,6 +62,37 @@ type EC2API interface {
 	WaitUntilInstanceStatusOk(input *ec2.DescribeInstanceStatusInput) error
 }
 
+func (u *updater) alreadyRunning(family string) (bool, error) {
+	log.Print("Checking for running updater tasks")
+	list, err := u.ecs.ListTasks(&ecs.ListTasksInput{
+		Cluster: &u.cluster,
+		Family:  aws.String(family),
+	})
+	if err != nil {
+		return false, fmt.Errorf("failed to list running updater tasks: %w", err)
+	}
+	if len(list.TaskArns) > 1 {
+		return true, nil
+	}
+	log.Println("This is the only running updater.")
+	return false, nil
+}
+
+func taskDefFamily() (string, error) {
+	taskDefARN := os.Getenv("TaskDefinitionARN")
+	taskDefARNType, err := arn.Parse(taskDefARN)
+	if err != nil {
+		return "", err
+	}
+	if !strings.Contains(taskDefARNType.Resource, "task-definition/") {
+		return "", fmt.Errorf("not a task definition arn: %q", taskDefARN)
+	}
+	// extract task definition family from resource: task-definition/<task definition family>:<revision>
+	family := strings.TrimPrefix(strings.SplitN(taskDefARNType.Resource, ":", 2)[0], "task-definition/")
```

Comment:
This seems a little backwards to me; breaking it out into a couple lines would make it more readable
```suggestion
	taskDef := strings.TrimPrefix(taskDefARNType.Resource, taskDefPrefix)
	family := strings.SplitN(taskDef, 2)[0]
```